router/sources/registration_tests.move

```diff
@@ -73,15 +75,58 @@ module router::registration_tests {
         let domain_name = utf8(b"test1");
 
         // Register with v1
-        router::register_domain(user, domain_name, SECONDS_PER_YEAR);
+        router::register_domain(user, domain_name, SECONDS_PER_YEAR, option::none(), option::none());
 
         // Bump mode
         router::set_mode(router, 1);
 
         // Fail to register with v2 because name is still active in v1
-        router::register_domain(user, domain_name, SECONDS_PER_YEAR);
+        router::register_domain(user, domain_name, SECONDS_PER_YEAR, option::none(), option::none());
+    }
+
+    #[test(
+        router = @router,
+        aptos_names = @aptos_names,
+        aptos_names_v2 = @aptos_names_v2,
+        user1 = @0x077,
+        user2 = @0x266f,
+        aptos = @0x1,
+        foundation = @0xf01d
+    )]
+    fun test_register_domain_with_target_addr_and_to_addr(
+        router: &signer,
+        aptos_names: &signer,
+        aptos_names_v2: &signer,
+        user1: signer,
+        user2: signer,
+        aptos: signer,
+        foundation: signer
+    ) {
+        router::init_module_for_test(router);
+        let users = test_helper::e2e_test_setup(aptos_names, aptos_names_v2, user1, &aptos, user2, &foundation);
+        let user1 = vector::borrow(&users, 0);
+        let user2 = vector::borrow(&users, 1);
+        let user1_addr = address_of(user1);
+        let user2_addr = address_of(user2);
+        let domain_name1 = utf8(b"test1");
+        let domain_name2 = utf8(b"test2");
+
+        // Register with v1
+        router::register_domain(user1, domain_name1, SECONDS_PER_YEAR, option::some(user2_addr), option::none());
+        assert!(router::is_name_owner(user1_addr, domain_name1, option::none()), 1);
+        assert!(*option::borrow(&router::get_target_addr(domain_name1, option::none())) == user2_addr, 2);
```

Comment:
also check primary name is unset now for user1? i think when user 1 set_target_address to user2, it will clear user1's primary name